### ISSUE TITLE
Django 4 compatibility

### ISSUE THREAD

**Issue by J4bbi** [yourlabs/django-session-security#145]
It's been a month since #142 was merged. Any chance a 2.7.0 release can be done?

Is there a concern that the changes might break anything? The code seems cautious.

I am looking to move to Django 4 and I would rather not have to modify my code manually.

**Comment by J4bbi** [comment]
@jpic , tagging you as you last committed

**Comment by AlexCLeduc** [comment]
For what it's worth to anyone else waiting on this, I've been using this monkey patch in my settings module, behaviour is identical to the merged PR

```python
import sys
from django.urls import re_path

class fakemodule(object):
    @staticmethod
    def url(*args, **kwargs):
        return re_path(*args, **kwargs)


sys.modules["django.conf.urls.defaults"] = fakemodule
```

**Comment by J4bbi** [comment]
@mpasternak my understanding is that the main branch is Django 4 compatible but a tagged release hasn't been done since [Jan 2020](https://github.com/yourlabs/django-session-security/releases/tag/2.6.6) and we need a tagged release to trigger a new build for pypi

**Comment by jpic** [comment]
@mpasternak <https://github.com/mpasternak> do you need any help to make a
pypi release? I can also make one if needed if course

**Comment by mpasternak** [comment]
@jpic hi! I'm on it, will let you know in a minute.

**Comment by mpasternak** [comment]
@jpic What I understand is I'm going to use either bumpversion or bumpver to tag 2.6.7 and then commit. 

I don't know if I have PyPI upload privileges. 

What's your preferred tool to upload stuff there? Twine? I'm experimenting with Poetry, looks promising. 

Another problem is that that Travis-CI stopped working and in the future it would be cool to move this project to GitHub Actions. I do have some small experience and I think I'll handle it.

**Comment by mpasternak** [comment]
@jpic @J4bbi I'll be hones with you guys, there's no Django 4.0 mention in tox.ini, first I'm going to fix/integrate CI

**Comment by jpic** [comment]
Can do a pre release meanwhile perhaps? So at least users can start
reporting problems they figure? Not sure what state the tests are in, might
need quite some upgrades in the test code

**Comment by mpasternak** [comment]
@jpic the plan is 
1) move to GitHub actions [done], 
2) make GitHub Actions pass [in progress]
3) add Django 4.0 to GitHub Actions
4) tag sources as 2.6.7-pre0, release
5) after 2 weeks, release 2.6.7 to PyPI

I am not sure if I have rights to write to PyPI but as it looks like you're around it should be good.

**Comment by jpic** [comment]
I'm alive and well but will miss some emails from time to time, on which
case you might want to reach me on YourLabs.chat

Unless you have any objection, I'd like to make a pre-release on pypi that
won't hit users who don't explicitly ask for the pre release in their
requirements, so that we can start getting more feedback about the version
in master, because I believe it can be helpful prior to make the actual
release that will have been through the new pipeline.

Keep up the good work 🎩

**Comment by mpasternak** [comment]
> 1. move to GitHub actions [done],
> 2. make GitHub Actions pass [in progress]
done!

> 3. add Django 4.0 to GitHub Actions
done! @jpic , take a look: https://github.com/yourlabs/django-session-security/actions/runs/2363443984

> 4. tag sources as 2.6.7-pre0, release
@jpic, did I understood that correctly? 

... or should I just edit the version and build package and send it to PyPI?

**Comment by mpasternak** [comment]
@jpic also please note that the method I used to move to GitHub Actions is the simplest method for the laziest of us and it uses ``tox.ini``. Chances are I'll be moving other of our yourlabs projects to GA

**Comment by jpic** [comment]
Mind blowing Michał!!

**Comment by mpasternak** [comment]
All set, tests pass, I need PyPI privileges @jpic 

```
Uploading distributions to https://upload.pypi.org/legacy/
Uploading django_session_security-2.6.7rc1-py3-none-any.whl
100% ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 123.5/123.5 kB • 00:00 • 81.5 MB/s
WARNING  Error during upload. Retry with the --verbose option for more details.
ERROR    HTTPError: 403 Forbidden from https://upload.pypi.org/legacy/
         The user 'dotz' isn't allowed to upload to project 'django-session-security'. See
         https://pypi.org/help/#project-name for more information.
```

**Comment by J4bbi** [comment]
@jpic, I think @mpasternak needs you to grant him PyPI privileges is possible

**Comment by mpasternak** [comment]
2.6.7-rc1 released

https://pypi.org/project/django-session-security/2.6.7rc1/

Feel free to test and let me know how's it going, let's give it a 2-week period, then I'm releasing final 2.6.7. 

Tips about release process kindly welcome.

**Comment by mpasternak** [comment]
Almost 2 weeks. Should I release the final one? I think I should

**Comment by mpasternak** [comment]
I have **no idea**, what I'm doing, but I released 2.6.7. Proceed with caution. Big thanks to everyone involved.
